hardware_interface/include/hardware_interface/components/component_info.hpp

```diff
@@ -46,14 +63,14 @@ struct ComponentInfo
    * \brief name of the command interfaces that can be set, e.g. "position", "velocity", etc.
    * Used by joints.
    */
-  std::vector<std::string> command_interfaces;
+  std::vector<InterfaceInfo> command_interfaces;
   /**
    * \brief name of the state interfaces that can be read, e.g. "position", "velocity", etc.
    * Used by Joints and Sensors.
    */
-  std::vector<std::string> state_interfaces;
+  std::vector<InterfaceInfo> state_interfaces;
   /**
-   * \brief (optional) key-value pairs of components parameters.
+   * \brief (optional) key-value pairs of components parameters, e.g. min_value and max_value.
    */
   std::unordered_map<std::string, std::string> parameters;
```

Comment:
Since the interfaces have their own attributes, should this `parameters` be deleted?

Reply:
We should leave it, since components could have some interface-unrelated parameters or also global min-max parameters (see [Example 1](https://github.com/ros-controls/roadmap/blob/master/design_drafts/components_architecture_and_urdf_examples.md#1-industrial-robots-with-only-one-interface))